libraries/AP_Notify/Display.cpp

```diff
@@ -297,11 +297,22 @@ bool Display::init(void)
     // initialise driver
     switch (pNotify->_display_type) {
         case DISPLAY_SSD1306:
-            _driver = new Display_SSD1306_I2C(hal.i2c_mgr->get_device(NOTIFY_DISPLAY_I2C_BUS, NOTIFY_DISPLAY_I2C_ADDR));
+            for(uint8_t i=0; i<3; i++) {
+                _driver = new Display_SSD1306_I2C(hal.i2c_mgr->get_device(i, NOTIFY_DISPLAY_I2C_ADDR));
+                if (_driver && _driver->hw_init()) {
```

Comment:
The probe stuff is nice in that you don't miss delete's accidentally.

However, doing this instance-method-hw_init thing allows you to start building up state in the created object, which is nice.

However - not trying to fix everything with this patch, just one thing :-)